frontend/src/metabase/admin/settings/components/Email/SettingsEmailForm.jsx

```diff
@@ -0,0 +1,61 @@
+import { Component } from "react";
+import { push } from "react-router-redux";
+import PropTypes from "prop-types";
+import { connect } from "react-redux";
+
+import { getSetting } from "metabase/selectors/settings";
+import { getIsEmailConfigured } from "metabase/setup/selectors";
+import { SettingsSection } from "../../app/components/SettingsEditor/SettingsSection";
+import { SMTPConnectionCard } from "./SMTPConnectionCard";
+
+class SettingsEmailForm extends Component {
+  static propTypes = {
+    elements: PropTypes.array.isRequired,
+    isEmailConfigured: PropTypes.bool,
+    isHosted: PropTypes.bool,
+    settingValues: PropTypes.object,
+    derivedSettingValues: PropTypes.object,
+    updateSetting: PropTypes.func,
+    reloadSettings: PropTypes.func,
+    push: PropTypes.func,
+  };
+
+  constructor(props, context) {
+    super(props, context);
+  }
+
+  render() {
+    const { elements, isHosted, isEmailConfigured, push } = this.props;
+
+    if (!isHosted && !isEmailConfigured) {
+      push("/admin/settings/email/smtp");
+    }
+
+    const settingElements = elements.filter(setting => !setting.getHidden?.());
+
+    return (
+      <>
+        {!isHosted && <SMTPConnectionCard />}
+        <SettingsSection
+          settingElements={settingElements}
+          settingValues={this.props.settingValues}
+          derivedSettingValues={this.props.derivedSettingValues}
+          updateSetting={this.props.updateSetting}
+          onChangeSetting={this.props.updateSetting}
+          reloadSettings={this.props.reloadSettings}
+        />
+      </>
+    );
+  }
+}
+
+const mapStateToProps = state => ({
+  isHosted: getSetting(state, "is-hosted"),
```

Comment:
I think it should be `is-hosted?` based on the BE code

Reply:
Since there are no test failures when it is `is-hosted` without the question mark, it would be nice to add a spec to ensure it works as expected on hosted instances

Reply:
Thanks for catching that!! :smile: 

Yeah, it should be `is-hosted?`. I never touched that line, so I never caught it. It must have just gotten out of date over time.

I added tests to check if the SMTP configuration is hidden or not based on `is-hosted?`